src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -153,6 +153,12 @@ type ClusterSpec struct {
 	DNSTrailingDotDisabled bool `json:"dnsTrailingDotDisabled,omitempty"`
 	// RestartConfig allows to control the behavior of the cluster when restarting
 	RestartConfig *RestartConfig `json:"restartConfig,omitempty"`
+	// Enable rack awareness for cluster. If enabled, the
+	// configurator will read each node's availability zone id
+	// label (topology.cloud.redpanda.com/zone-id) and assign it as the
+	// rack of the corresponding redpanda instance. If that label
+	// is missing it will use (topology.kubernetes.io/zone).
+	RackAwareness bool `json:"rackAwareness,omitempty"`
```

Comment:
question: why is this not just the default. and we can let redpanda choose if single zone or multi zone, but from the POV of the operator, shouldn't it always pass the k8s zone?

Reply:
The *only* place where `RackAwareness` has an effect is in creating the k8s ServiceAccount that is used for the configurator to talk to the k8s api server (and retrieve the zone label). If not for that, we could just provide the rack to Redpanda and be empty in cases where the node's zone is empty, e.g., running locally or not running on a cloud provider.

I'll look into whether there's a concern with creating the ServiceAccount regardless. In the existing code (pre this PR) we only create it when using external connectivity. But I'm not seeing why at the moment.

Reply:
Does Redpanda accept all `rack` properties being empty when `enable_rack_awareness==true`?

Reply:
yes, missing the `rack` field is acceptable (or being an empty string, but we're going with the former, ommitting).